src/participant.rs

```diff
@@ -45,7 +46,22 @@ pub(crate) enum ProcessOutcome<O> {
     Terminated(O),
 }
 
-impl<O> ProcessOutcome<O> {
+impl<O> Debug for ProcessOutcome<O> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let descriptor = match self {
+            ProcessOutcome::Incomplete => "Incomplete",
+            ProcessOutcome::Processed(_) => "Processed",
+            ProcessOutcome::Terminated(_) => "Terminated",
+            ProcessOutcome::TerminatedForThisParticipant(_, _) => "Terminated for this participant",
+        };
+        write!(f, "ProcessOutcome::{descriptor}")
+    }
+}
+
+impl<O> ProcessOutcome<O>
+where
+    O: std::fmt::Debug,
```

Comment:
Probably doesn't matter, but this trait bound sticks out to me `O: std::fmt::Debug`. Why is it necessary?

Reply:
Oh NVM. I see how it is used for error logging.

Reply:
I agree that it's kind of awkward.